### PR TITLE
Update CI to use gcc-14

### PR DESCRIPTION
**Pull request by derobins**

### CODE REVIEW

**Comment by derobins** [comment]
I'll probably do the warnings in separate PRs before this will be ready to merge

**Review by bmribler — APPROVED** [review]

**Review by qkoziol — APPROVED** [review]

**Review by bmribler — APPROVED** [review]

File: hl/src/H5TB.c [hunk 1]
```diff
@@ -3185,7 +3185,7 @@ H5TB_create_type(hid_t loc_id, const char *dset_name, size_t type_size, const si
     if (H5TBget_table_info(loc_id, dset_name, &nfields, NULL) < 0)
         goto out;
 
-    if (NULL == (fnames = (char **)calloc(sizeof(char *), (size_t)nfields)))
+    if (NULL == (fnames = (char **)calloc((size_t)nfields), sizeof(char *)))
```
Comment (lrknox) [inline hl/src/H5TB.c]:
```suggestion
    if (NULL == (fnames = (char **)calloc((size_t)nfields, sizeof(char *))))
```

Comment (lrknox) [reply hl/src/H5TB.c]: Misplaced ')' causes build error.

File: hl/src/H5TB.c [hunk 2]
```diff
@@ -3185,7 +3185,7 @@ H5TB_create_type(hid_t loc_id, const char *dset_name, size_t type_size, const si
     if (H5TBget_table_info(loc_id, dset_name, &nfields, NULL) < 0)
         goto out;
 
-    if (NULL == (fnames = (char **)calloc(sizeof(char *), (size_t)nfields)))
+    if (NULL == (fnames = (char **)calloc((size_t)nfields, sizeof(char *)))
```
Comment (lrknox) [inline hl/src/H5TB.c]:
```suggestion
    if (NULL == (fnames = (char **)calloc((size_t)nfields, sizeof(char *))))
```

File: .github/workflows/abi-report.yml [hunk 3]
```diff
@@ -28,7 +28,7 @@ permissions:
 
 jobs:
   check:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-24.04
```
Comment (byrnHDF) [inline .github/workflows/abi-report.yml:31]: Should we not specify latest and always use a specific version on all CI?

**Review by bmribler — APPROVED** [review]

File: .github/workflows/abi-report.yml (hunk 3 above)
Comment (derobins) [reply .github/workflows/abi-report.yml:31]: I definitely prefer that, but ubuntu-latest is 22.04, which does not have gcc-14

**Comment by derobins** [comment]
I'll recreate when we've fixed the warning issues

**Pull request closed without merging**